### ISSUE TITLE
this.Success is not a function

### ISSUE THREAD

**Issue by ccorcos** [folktale/data.validation#10]
https://github.com/folktale/data.validation/blob/master/lib/validation.js#L141

**Comment by davidchambers** [comment]
It looks as though it should be `Validation.Success` rather than `this.Success`.

**Comment by robotlolita** [comment]
JS does not auto-bind methods when you select them from an object, so:

```js
var f = Validation.of;
f("something")
```

Will be equivalent to:

```
Validation.of.call(undefined, "something")
// or, in non-strict mode:
Validation.of.call(global, "something")
```

Neither the global object, nor the undefined value have a `Success` property (usually), so it fails.

To pass the function around, right now, you would need to bind it:

```js
[1, 2, 3].map(Validation.Success.bind(Validation))
```

Or use an arrow function:

```js
[1, 2, 3].map(x => Validation.Success(x))
```

This is an annoying misfeature of JavaScript, and I still haven't decided how to deal with it. I *could* change `Validation.Success` to be static, so it'd basically have the same code as `Validation.of`, however, that would be inconsistent with things like `Validation.ap` which need to work on a validation instance because it needs to get the `value` from somewhere. I could change all functions to be static, so nothing would use this, but then it would be slow and not-optimisable, because each instance would yield a different hidden class/type shape in v8/SpiderMonkey/etc, which in turn would make every function megamorphic and slow down all the things that Validation touches. Neither look like good solutions to me :/

**Comment by robotlolita** [comment]
This is fixed as of 1.3.1. All methods directly on the Validation object are now static functions.

Sorry for the long time it took to fix this, and thanks for the PR :)
